portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,49 +16,69 @@
 
 import com.google.sps.data.Comment;
 import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns and saves comments in Datastore */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-  private ArrayList<Comment> commentList;
-
-  @Override
-  public void init() {
-    commentList = new ArrayList<Comment>();
-  }
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    json = convertToJson(commentList);
+    int commentLimit = Integer.parseInt(request.getParameter("limit"));
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Can this be a class member which gets initialized in init()?

Reply:
I added a commit which initializes datastore in init() for both DataServlet.java and DeleteCommentServlet.java.

Reply:
So I changed it back to having the datastore initialization inside the functions because of Linzi's suggestion that a servlet must handle concurrent requests and that it would be better to have one DatastoreService object per goGet/doPost request.